docs/manual/guide/services/HeaderFilters.md

```diff
@@ -0,0 +1,30 @@
+# Header Filters
+
+In Lagom you may add `HeaderFilter`s to your `ServiceDescriptor`. In a `HeaderFilter` you will usually handle protocol negotiation or authentication. 
+
+A single `HeaderFilter` implementation may transform a request leaving a client or entering a server and a response leaving a server and entering a client.
+
+@[user-agent-auth-filter](code/docs/services/UserAgentHeaderFilter.java)
```

Comment:
I believe it can pull snippets from the main code base if you give a relative path to it. It does that with a bunch of `reference.conf` files already (see `PersistentEntityRDBMS.md` for example). So there shouldn't be a need to duplicate the `UserAgentHeaderFilter` code.